### PR TITLE
Add upgrade tests for stable 4.14 to stable 4.15

### PR DESCRIPTION
**Pull request by liangxia**

### CODE REVIEW

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-stable-4.15-upgrade-from-stable-4.14-gcp-upi-f28

**Comment by liangxia** [comment]
Got the expected result,
```
INFO[2024-01-02T09:13:59Z] Resolved release latest to quay.io/openshift-release-dev/ocp-release:4.14.7-x86_64 
...
INFO[2024-01-02T09:13:59Z] Resolved release target to quay.io/openshift-release-dev/ocp-release:4.15.0-rc.0-x86_64
```

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__arm64-stable-4.15-upgrade-from-stable-4.14.yaml [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+      version_bounds:
+        lower: 4.15.0-0
+        upper: 4.16.0-0
+resources:
+  '*':
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tests:
+- as: aws-ipi-byo-iam-role-f28
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__arm64-stable-4.15-upgrade-from-stable-4.14.yaml:78]: This yaml should be copied from `openshift-openshift-tests-private-release-4.14__arm64-stable-4.14-upgrade-from-stable-4.13.yaml`, then update the version setting, but not copied from nightly-upgrade-from-stable, right?

Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__arm64-stable-4.15-upgrade-from-stable-4.14.yaml:78]: It depends on how many profiles do we want to cover in stable to stable upgrade tests, full profile coverage or partial profile coverage ?
I'm fine with either one. Which one do you prefer ?

Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__arm64-stable-4.15-upgrade-from-stable-4.14.yaml:78]: I prefer partial profile coverage, the stable to stable upgrade tests are selected on some specific install profiles.

Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__arm64-stable-4.15-upgrade-from-stable-4.14.yaml:78]: Ack

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-stable-4.15-upgrade-from-stable-4.14-aws-ipi-ovn-fips-f28

**Comment by liangxia** [comment]
Working as expected,
```
INFO[2024-01-02T09:54:17Z] Resolved release latest to quay.io/openshift-release-dev/ocp-release:4.14.7-x86_64 
...
INFO[2024-01-02T09:54:17Z] Resolved release target to quay.io/openshift-release-dev/ocp-release:4.15.0-rc.0-x86_64 
```

**Pull request merged**